src/BridgedRoom.ts

```diff
@@ -474,7 +475,10 @@ export class BridgedRoom {
             const ghost = await this.main.getGhostForSlackMessage(message, this.slackTeamId!);
             await ghost.update(message, this);
             await ghost.cancelTyping(this.MatrixRoomId); // If they were typing, stop them from doing that.
-            return await this.handleSlackMessage(message, ghost, content);
+            this.slackSendLock = this.slackSendLock.finally(async () => {
```

Comment:
Does this leak promises that have finished?

Reply:
The following line should take care of it